os/Makefile.unix

```diff
@@ -495,7 +475,58 @@ ifeq ($(CONFIG_RAW_BINARY),y)
 	$(Q) $(OBJCOPY) $(OBJCOPYARGS) -O binary $(BIN) $(BIN).bin
 endif
 
+define VERIFY_APP
+	$(Q) echo "Verify $(OUTBIN_DIR)/$1"
+	$(Q) if [ "`nm -u $(OUTBIN_DIR)/$1 | wc -l`" != "0" ]; then \
+		echo "Undefined Symbols in $(OUTBIN_DIR)/$1"; \
+		nm -u -l $(OUTBIN_DIR)/$1; \
+		rm $(OUTBIN_DIR)/$1; \
+		exit 1; \
+	fi
+endef
+
+ifeq ($(CONFIG_ELF_EXCLUDE_SYMBOLS),y)
+define STRIP_DBG
+	$(Q) $(OBJCOPY) --remove-section .comment $(OUTBIN_DIR)/$1
+	$(Q) $(STRIP) -g $(OUTBIN_DIR)/$1 -o $(OUTBIN_DIR)/$1
+endef
+endif
+
+ifeq ($(CONFIG_COMPRESSED_BINARY),y)
+define COMPRESS_BIN
+	$(Q) cp $(OUTBIN_DIR)/$1 $(OUTBIN_DIR)/$1.uncomp
+	$(Q) tools/compression/mkcompressimg $(CONFIG_COMPRESSION_BLOCK_SIZE) $(CONFIG_COMPRESSION_TYPE) $(OUTBIN_DIR)/$1 $(OUTBIN_DIR)/$1.comp
+	$(Q) mv $(OUTBIN_DIR)/$1.comp $(OUTBIN_DIR)/$1
+endef
+endif
+
+define PREPARE_APP
```

Comment:
Could you add description of each define?

Reply:
Done.